Wizardry/Core/BattleSys/Source/BattleCalcReal.c

```diff
@@ -131,6 +131,14 @@ STATIC_DECLAR void BattleCalcReal_ComputSkills(struct BattleUnit * attacker, str
             attacker->battleAttack = 0;
     }
 #endif
+
+#if (defined(SID_NoGuard) && (SID_NoGuard < MAX_SKILL_NUM))
+        if (SkillTester(&attacker->unit, SID_NoGuard))
+        {
+            attacker->battleEffectiveHitRate = 100;
+            attacker->battleAvoidRate = -100;
```

Comment:
after checking on vanilla, `BattleUnit::battleAvoidRate` is useless at that timing, there has been no reason to change this value